### PR TITLE
Add OpenCover support to Bozo

### PR DESCRIPTION
**Pull request by lavapatch**
Add ability to generate code coverage reports using OpenCover.

http://youtrack.zopa.local/issue/PT-6

To enable change the nunit section of  the bozorc.rb of C# project from

```ruby
test_with :nunit do |n|
  n.platform 'x86'
  n.project 'TEST_PROJECT_NAME_GOES_HERE'
end
```

to 

```ruby
test_with :open_cover do |d|
  d.runner :nunit do |n|
    n.platform 'x86'
    n.project 'TEST_PROJECT_NAME_GOES_HERE'
  end
end
```

### CODE REVIEW

File: lib/bozo/test_runners/opencover.rb [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      if ENV['openCover.home'].nil?
+        if ENV['ProgramFiles(x86)'].nil?
+          program_files_path = ENV['ProgramFiles']
+        else
+          program_files_path = ENV['ProgramFiles(x86)']
+        end
+
+        File.join(program_files_path, 'OpenCover', 'OpenCover.Console.exe')
+      else
+        File.join(ENV['teamcity.dotCover.home'], 'OpenCover.Console.exe')
```
Comment (stephenbinns) [inline lib/bozo/test_runners/opencover.rb]: `teamcity`?

**Comment by stephenbinns** [comment]
Can we also add a tools configuration similar to what we do for nuget? https://github.com/zopaUK/bozo-scripts/blob/master/lib/bozo/tools/nuget.rb

**Comment by slavo** [comment]
Nice!

I wonder if we can integrate with Jenkins and show badges on Github like so?

buildpassbadges

**Comment by stephenbinns** [comment]
@slavo we've planning on using this with sonarqube to produce these reports which can be integrated somewhat http://docs.sonarqube.org/display/PLUG/GitHub+Plugin

**Comment by slavo** [comment]
@stephenbinns SonarQube is GPL (the code we use it in has to be public domain). Does this affect us?

**Comment by lavapatch** [comment]
No. GPL pertains to changes made to the tool/library itself

**Comment by slavo** [comment]
@lavapatch Yes, but it's copyleft (https://en.wikipedia.org/wiki/Copyleft).

"with the stipulation that the same rights be preserved in derivative works down the line".

This means our code has to be licensed under GPL, too.

**Comment by lavapatch** [comment]
@slavo We are using the tool As Is. We don't need to worry about GPL is were are not redistributing the tool nor are we modifying or deriving anything from it.

**Comment by stephenbinns** [comment]
LGTM :+1:

**Pull request merged**